src/detect-engine-hsbd.c

```diff
@@ -270,12 +299,20 @@ int DetectEngineInspectHttpServerBody(ThreadVars *tv,
     HtpState *htp_state = (HtpState *)alstate;
     uint32_t buffer_len = 0;
     uint32_t stream_start_offset = 0;
-    const uint8_t *buffer = DetectEngineHSBDGetBufferForTX(tx, tx_id,
-                                                     de_ctx, det_ctx,
-                                                     f, htp_state,
-                                                     flags,
-                                                     &buffer_len,
-                                                     &stream_start_offset);
+    const uint8_t *buffer = NULL;
+
+    if (det_ctx->decompressed_hsbd_buffer) {
+        buffer = det_ctx->decompressed_hsbd_buffer;
```

Comment:
Can you look at how DetectEngineHSBDGetBufferForTX manages it's buffers. It uses an array in det_ctx that contains the actual buffers so that we can have one per tx (in case we inspect multiple txs at the same time). It automatically makes sure that 2+ calls result in only one buffer creation, the next calls just use the cached results. I would like the flash functionality to use this same logic.